File: bainary_search.cpp
#include <bits/stdc++.h>
using namespace std;
int main(int argc, char const *argv[])
{
    int n = 9, left = 0, right = n, mid = 0, search_item = 90;
    int arr[10] = {10, 20, 30, 40, 50, 60, 70, 80, 90, 100};

    while (left <= right)
    {
        mid = (left + right) / 2;
        if (arr[mid] == search_item)
        {
            printf("\nthe value of %d at index %d\n", arr[mid], mid);
            return 0;
        }
        else if (arr[mid] > search_item)
        {
            right = mid - 1;
        }
        else
        {
            left = mid + 1;
        }
    }
    printf("\nValue not found !\n");

    return 0;
}

File: Bainary_search.py
arr =[2,8,11,15,19,22,27,30,35,45]
target = 35
s=0
e=len(arr)-1
# print(e)
for i in arr:
    mid = (s+e)//2
    # print(arr[mid])
    if(arr[mid]==target):
        print("target found at index : ",mid)
        break
    elif(mid<target):
        right = mid+1
        s=right
    elif(mid>target):
        left = mid-1
        e = left        